visidata/loaders/geojson.py

```diff
@@ -0,0 +1,130 @@
+
+        self.refresh()
+
+    def parse_geometry(self, row, colour, bbox=None):
+        if bbox is None: bbox = [180, 90, -180, -90]
+
+        typ = row['geometry']['type']
+        if typ == 'GeometryCollection':
+            for g in row['geometries']:
+                bbox = self.parse_geometry(row, colour, bbox)
+            return bbox
+
+        coords = row['geometry']['coordinates']
+        if typ in ('Point', 'LineString', 'Polygon'):
+            coords = [coords]
+
+        if typ in ('Point', 'MultiPoint'):
+            for x, y in coords:
+                self.point(x, y, colour, row)
+            bbox = reduce_coords(coords, bbox)
+        elif typ in ('LineString', 'MultiLineString'):
+            for line in coords:
+                self.polyline(line, colour, row)
+                bbox = reduce_coords(line, bbox)
+        elif typ in ('Polygon', 'MultiPolygon'):
+            for polygon in coords:
+                self.polygon(polygon[0], colour, row)
+                bbox = reduce_coords(polygon[0], bbox)
+                for hole in polygon[1:]:
+                    self.polygon(hole, 0, row)
+        else:
+            vd.status('notimpl shapeType %s' % typ)
+
+        return bbox
+
+def reduce_coords(coords, initial):
+    return reduce(
+        lambda a,n: [min(a[0],n[0]), min(a[1],n[1]), max(a[2],n[0]), max(a[3],n[1])],
+        coords, initial)
+
+@VisiData.api
+def save_geojson(vd, p, vs):
+    if (isinstance(vs, ShapeMap)):
+        return save_shp_geojson(vd, p, vs)
+
+    isinstance(vs, Canvas) or vd.fail("must save GeoJSON from map sheet")
+    features = []
+    keyCols = list(map(lambda c: c.name, vs.source.keyCols))
+    for row in Progress(vs.sourceRows, 'saving'):
+        row = deepcopy(row)
+        row['properties'] = {k:v for k,v in row.get('properties', {}).items() if k in keyCols}
+        features.append(row)
+
+    featcoll = {
+        'type': 'FeatureCollection',
+        'features': features,
+    }
+
+    try:
+        indent = int(vs.options.json_indent)
+    except Exception:
+        indent = vs.options.json_indent
+
+    with p.open_text(mode='w') as fp:
+        encoder = json.JSONEncoder(indent=indent, sort_keys=vs.options.json_sort_keys)
+        for chunk in encoder.iterencode(featcoll):
+            fp.write(chunk)
+
+GeoJSONSheet.addCommand('.', 'plot-row', 'vd.push(GeoJSONMap(name+"_map", sheet, sourceRows=[cursorRow], textCol=cursorCol, source=sheet))', 'plot geospatial vector in current row')
+GeoJSONSheet.addCommand('g.', 'plot-rows', 'vd.push(GeoJSONMap(name+"_map", sheet, sourceRows=rows, textCol=cursorCol, source=sheet))', 'plot all geospatial vectors in current sheet')
+InvertedCanvas.addCommand('^S', 'save-sheet', 'vd.saveSheets(inputPath("save to: ", value=getDefaultSaveName(sheet)), sheet, confirm_overwrite=options.confirm_overwrite)', 'save current sheet to filename in format determined by extension (default .tsv)')
```

Comment:
Maybe ultimately this should go on BaseSheet, but for now I think this should be on GeoJSONSheet specifically and not all InvertedCanvas.

Reply:
GeoJSONSheet already has a save-sheet on it, do you mean GeoJSONMap?